client/alloc_runner.go

```diff
@@ -28,6 +28,12 @@ const (
 // AllocStateUpdater is used to update the status of an allocation
 type AllocStateUpdater func(alloc *structs.Allocation)
 
+// AllocStatsReporter exposes stats related APIs of an allocation runner
+type AllocStatsReporter interface {
+	AllocStats() map[string]TaskStatsReporter
+	TaskStats(task string) (TaskStatsReporter, error)
```

Comment:
Seems like you can get rid of `TaskStats` and just do a map look up by the caller. Also add comments on the method